File: scratch/python/fetch.py
import httpx

def fetch_capabilities(api_key: str) -> httpx.Response:
    with httpx.Client(http1=True, http2=False) as client:
        response = client.get(
            "https://public-api.meteofrance.fr/public/arome/1.0/wcs/MF-NWP-HIGHRES-AROME-001-FRANCE-WCS/GetCapabilities",
            params={"service": "WCS", "version": "2.0.1", "language": "eng"},
            headers={
                "accept": "*/*",
                "apikey": api_key,
            },
            timeout=30,
        )
    return response

def fetch_describe_coverage(api_key: str, coverage_id: str) -> httpx.Response:
    with httpx.Client(http1=True, http2=False) as client:
        response = client.get(
            "https://public-api.meteofrance.fr/public/arome/1.0/wcs/MF-NWP-HIGHRES-AROME-001-FRANCE-WCS/DescribeCoverage",
            params={"service": "WCS", "version": "2.0.1", "coverageID": coverage_id},
            headers={
            "accept": "*/*",
            "apikey": api_key,
        },
        timeout=30,
    )
    return response

def fetch_coverage(api_key: str, coverage_id: str, subset: list[str]) -> httpx.Response:
    with httpx.Client(http1=True, http2=False) as client:
        response = client.get(
            "https://public-api.meteofrance.fr/public/arome/1.0/wcs/MF-NWP-HIGHRES-AROME-001-FRANCE-WCS/GetCoverage",
            params={
                "service": "WCS",
                "version": "2.0.1",
                "coverageID": coverage_id,
                "subset": subset,
                "format": "image/tiff",
                },
            headers={
                "accept": "*/*",
                "apikey": api_key,
            },
            timeout=30,
        )
    return response

File: src/lib/wind/fetch.ts
import axios, { type AxiosResponse } from "axios";

const BASE_URL =
  "https://public-api.meteofrance.fr/public/arome/1.0/wcs/MF-NWP-HIGHRES-AROME-001-FRANCE-WCS";

export async function fetchCapabilities(apiKey: string) {
  return axios.get(`${BASE_URL}/GetCapabilities`, {
    params: { service: "WCS", version: "2.0.1", language: "eng" },
    headers: { accept: "*/*", apikey: apiKey },
    timeout: 30000,
  });
}

export async function fetchDescribeCoverage(apiKey: string, coverageId: string) {
  return axios.get(`${BASE_URL}/DescribeCoverage`, {
    params: { service: "WCS", version: "2.0.1", coverageID: coverageId },
    headers: { accept: "*/*", apikey: apiKey },
    timeout: 30000,
  });
}

export async function fetchCoverage(
  apiKey: string,
  coverageId: string,
  subset: string[]
): Promise<AxiosResponse> {
  // Construire l'URL manuellement pour éviter subset[]=...
  const params = new URLSearchParams({
    service: "WCS",
    version: "2.0.1",
    coverageID: coverageId,
    format: "image/tiff",
  });

  // Ajouter chaque subset séparément → &subset=...&subset=...
  for (const s of subset) {
    params.append("subset", s);
  }

  const url = `${BASE_URL}/GetCoverage?${params.toString()}`;

  return axios.get(url, {
    headers: { accept: "*/*", apikey: apiKey },
    timeout: 30000,
    responseType: "arraybuffer",
  });
}